### PR TITLE
Lifecycle batch 1

### PR DESCRIPTION
**Pull request by alexduf**
## What does this change?

This PR starts to refactor our lifecycle-related events (onStart, onStop) into injectable objects. These injectable object are necessary to be able to migrate to play 2.5.
Given the sheer size of the full change (600+ lines) I started re-deviding it in 3 PR. This is the first batch, expect two more to come on that topic only.
Because of that, a temporary class `BackwardCompatibleLifecycleComponents` makes the link between the old world of `Global`s and the world of `ApplicationLoaders`.
## What is the value of this and can you measure success?

Smooth migration towards play2.5, human size PR
## Request for comment

@TBonnin @johnduffell

### CODE REVIEW

File: archive/app/services/ArchiveMetrics.scala [hunk 1]
```diff
@@ -1,32 +1,32 @@
...
+  appLifecycle.addStopHook{ () => Future{
+    Jobs.deschedule("ArchiveSystemMetricsJob")
+  }}
 
   private def report() {
     CloudWatch.putMetrics("ArchiveMetrics", List(GoogleBotMetric.Googlebot404Count), List.empty)
   }
 
-  override def onStart(app: PlayApp) {
+  override def start():Unit = {
```
Comment (johnduffell) [inline archive/app/services/ArchiveMetrics.scala]: v minor but I'd have added a space before Unit after the :

**Comment by johnduffell** [comment]
looks good, well done on going through all that, pretty mechanical stuff 👍

File: admin/app/Global.scala [hunk 2]
```diff
@@ -1,24 +1,31 @@
...
   with SurgingContentAgentLifecycle
   with DfpAgentLifecycle
-  with DfpDataCacheLifecycle
-  with Logstash
-  with AdminHealthCheckLifeCycle {
+  with Logstash {
 
   override lazy val applicationName = "frontend-admin"
+
+  def lifecycleComponents(appLifecycle: ApplicationLifecycle)(implicit ec: ExecutionContext): List[LifecycleComponent] = List(
```
Comment (TBonnin) [inline admin/app/Global.scala]: missing override?

**Comment by TBonnin** [comment]
👍  thanks again for doing this

**Pull request merged**